### ISSUE TITLE
disallow right-hand side values TypedDict

### ISSUE THREAD

**Issue by vpoverennov** [astral-sh/ruff#17225]
### Summary

```
from typing import TypedDict


class A(TypedDict):
    x: int = 42
```

This is not allowed according to [PEP 589](https://peps.python.org/pep-0589/)
> TypedDict doesn’t support providing a default value type for keys that are not explicitly defined. This would allow arbitrary keys to be used with a TypedDict object, and only explicitly enumerated keys would receive special treatment compared to a normal, uniform dictionary type.

And is indeed ignored at runtime if you instantiate the class
```
>>> A()
{}
```

mypy:
```
y.py:5: error: Right hand side values are not supported in TypedDict  [misc]
```

pyright:
```
y.py:5:5 - error: TypedDict classes can contain only type annotations (reportGeneralTypeIssues)
```

Would be curious to get your thoughts for such a rule, thank you for making the awesome tool

**Comment by karthikeya9999** [comment]
from typing import TypedDict

class A(TypedDict, total=False):
    x: int

a: A = {}  # x is optional
b: A = {'x': 42}  # okay too

**Comment by MichaReiser** [comment]
@karthikeya9999 

I'm not sure I understand your comment. Is it in favor of the rule or an argument against it?

**Comment by Sriketk** [comment]
I second this.

```py
class State(MessagesState):
    """Extended MessagesState with medical case information."""

    context: str
    question: str
    answer: str
    userAnswer: str = ""
    questionAnswered: bool = False

    # Optional medical information
    medications: list[str] = []
    allergies: list[str] = []
    familyHistory: list[str] = []
    labResults: list[str] = []
    options: list[str] = []
    bloodPresure: str = ""
    respirations: str = ""
    pulse: str = ""
    physicalExamination: str = ""
    temperature: str = ""
    history: str = ""
    demographics: str = ""


graph_builder = StateGraph(State)
```

linter is giving me error here 

```
src/agent/graph.py:117: error: Right hand side values are not supported in TypedDict  [misc]
```

**Comment by MichaReiser** [comment]
Given that mypy and pyright already support this. I feel like it's less urgent for ruff to add this rule now (as it also depends on multifile analysis to resolve if the class extends a typed dict)

**Comment by vpoverennov** [comment]
caught by `ty` - https://play.ty.dev/d0c684e2-98f8-4e51-9ad0-cccfd5b6453a

```
TypedDict item cannot have a value (invalid-typed-dict-statement) [Ln 5, Col 14]
```
